src/idl_gen_cpp.cpp

```diff
@@ -2627,7 +2636,7 @@ class CppGenerator : public BaseGenerator {
             break;
           }
           default: {
-            if (field.value.type.enum_def) {
+            if (field.value.type.enum_def && !opts_.scoped_enums) {
```

Comment:
This condition is the same as `opts_.scoped_enums && IsEnum(type);` at line 632, isn't?